public/app/plugins/datasource/streaming/query_ctrl.ts

```diff
@@ -0,0 +1,53 @@
+import appEvents from 'app/core/app_events';
+import { QueryCtrl } from 'app/plugins/sdk';
+
+export class StreamingQueryCtrl extends QueryCtrl {
+  static templateUrl = 'partials/query.editor.html';
+
+  dataPreview: any;
+
+  /** @ngInject */
+  constructor($scope: any, $injector: any) {
+    super($scope, $injector);
+
+    appEvents.on('ds-request-response', this.onResponseReceived, $scope);
+    this.panelCtrl.events.on('refresh', this.onRefresh, $scope);
+    this.panelCtrl.events.on('data-received', this.onDataReceived, $scope);
+  }
+
+  // Called from the search window
+  onKeyPress(event: any) {
+    if (event.which === 13) {
+      this.panelCtrl.refresh();
+    }
+  }
+
+  onDataReceived = dataList => {
```

Comment:
this is the only (I think) remaining puzzle piece for react query editors. Just gave it a quick stab: https://github.com/grafana/grafana/pull/16233

Reply:
this particular line is just copy/paste.  The streaming server config options need a lot of love